onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -81,38 +81,56 @@ def aten_add(self: TReal, other: TReal, alpha: float = 1.0) -> TReal:
     return op.Add(self, other)
 
 
+@torch_op("aten::addbmm")
 def aten_addbmm(
-    self: TensorType,
-    batch1: TensorType,
-    batch2: TensorType,
+    self: TReal,
+    batch1: TReal,
+    batch2: TReal,
     beta: float = 1.0,
     alpha: float = 1.0,
-) -> TensorType:
-    """addbmm(Tensor self, Tensor batch1, Tensor batch2, *, Scalar beta=1, Scalar alpha=1) -> Tensor"""
+) -> TReal:
+    """addbmm(Tensor self, Tensor batch1, Tensor batch2, *, Scalar beta=1, Scalar alpha=1) -> Tensor
 
-    raise NotImplementedError()
+    Performs a batch matrix-matrix product of matrices stored in `batch1` and `batch2`,
+    with a reduced add step (all matrix multiplications get accumulated along the first
+    dimension). `self` is added to the final result.
 
+    `batch1` and `batch2` must be 3-D tensors each containing the same number of matrices.
+    """
 
-def aten_addcdiv(
-    self: TensorType, tensor1: TensorType, tensor2: TensorType, value: float = 1.0
-) -> TensorType:
-    """addcdiv(Tensor self, Tensor tensor1, Tensor tensor2, *, Scalar value=1) -> Tensor"""
+    scaled_self = op.Mul(self, beta)
+    axes = op.Constant(value_ints=[0])
+    reduced_batches = op.ReduceSum(op.MatMul(batch1, batch2), axes=axes, keepdims=False)
```

Comment:
Why do we need axes=axes for an input parameter?

Reply:
Do you suggest replacing it with just `op.Constant(value_ints=[0])`, or something else?

Reply:
op.ReduceSum(op.MatMul(batch1, batch2), axes, keepdims=False) ?

Reply:
Done